yjit.rb

```diff
@@ -44,11 +44,18 @@ def self.reset_stats!
   #     * `false`: Don't enable the log.
   #     * `true`: Enable the log. Print log at exit.
   #     * `:quiet`: Enable the log. Do not print log at exit.
-  def self.enable(stats: false, log: false)
+  def self.enable(stats: false, log: false, exec_mem_size: nil, call_threshold: nil)
     return false if enabled?
-    at_exit { print_and_dump_stats } if stats
-    call_yjit_hooks
-    Primitive.rb_yjit_enable(stats, stats != :quiet, log, log != :quiet)
+
+    if exec_mem_size.nil? && call_threshold.nil?
+      # Proceed with original enable process
+      at_exit { print_and_dump_stats } if stats
+      call_yjit_hooks
+      Primitive.rb_yjit_enable(stats, stats != :quiet, log, log != :quiet)
+    else
+      # Call helper function for custom configuration
+      enable_with_custom_config(stats, log, exec_mem_size, call_threshold)
+    end
```

Comment:
This seems a bit complicated.

How about passing the Ruby values directly to `Primitive.rb_yjit_enable` ?